### ISSUE TITLE
TypeScript definitions for `min()`/`max()` functions return `any`

### ISSUE THREAD

**Issue by Maxim-Mazurok** [josdejong/mathjs#2954]
**Describe the bug**
These are type definitions of min and max functions:
```ts
// eslint-disable-next-line @typescript-eslint/no-explicit-any
max(...args: MathType[]): any
//...
// eslint-disable-next-line @typescript-eslint/no-explicit-any
min(...args: MathType[]): any
```
So whenever I try to use their result in my own code - it triggers `@typescript-eslint/no-unsafe-argument` error.

**To Reproduce**
`const result = max(1, 2);`

**Potential fix**
Currently, I'm using the following patch which seems reasonable to me:
```patch
diff --git a/node_modules/mathjs/types/index.d.ts b/node_modules/mathjs/types/index.d.ts
index e7d7e21..3bfa799 100644
--- a/node_modules/mathjs/types/index.d.ts
+++ b/node_modules/mathjs/types/index.d.ts
@@ -2596,8 +2596,7 @@ declare namespace math {
      * @param args A single matrix or multiple scalar values
      * @returns The maximum value
      */
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    max(...args: MathType[]): any
+    max<T extends MathType[]>(...args: T): T[number]
     /**
      * @param A A single matrix
      * @param dim The maximum over the selected dimension
@@ -2645,8 +2644,7 @@ declare namespace math {
      * @param args A single matrix or or multiple scalar values
      * @returns The minimum value
      */
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    min(...args: MathType[]): any
+    min<T extends MathType[]>(...args: T): T[number]
     /**
      * @param A A single matrix
      * @param dim The minimum over the selected dimension
```

PR with the fix: https://github.com/josdejong/mathjs/pull/2955

**Comment by josdejong** [comment]
Fixed via #2955, thanks Maxim.

**Comment by josdejong** [comment]
Fix published now in `v11.8.1`, thanks again.

**Comment by Maxim-Mazurok** [comment]
I can confirm it's working now, thank you!
